src/Database/Rows.php

```diff
@@ -132,6 +132,40 @@ public function clear()
 		$this->rows = array();
 	}
 
+	/**
+	 * Selects a number of randomly selected rows
+	 *
+	 * @param   integer  $n  The number of rows to randomly select
+	 * @return  Rows
+	 * @since   2.1.13
+	 **/
+	public function pickRandom($n)
+	{
+		$rows = $this->rows;
+
+		shuffle($rows);
+		$randomRows = array_slice($rows, 0, $n);
+		$rowsObject = new self();
+		$rowsObject->_setRows($randomRows);
+
+		return $rowsObject;
+	}
+
+	/**
+	 * Set the array of rows
+	 *
+	 * @param   array  $rows  The array to set the $rows attribute to
+	 * @return  null
+	 * @since   2.1.13
+	 **/
+	protected function _setRows($rows)
```

Comment:
My thought with this is that I noticed the class doesn't have a constructor, so maybe:
```
public function __construct($rows = array())
{
    foreach ((array)$rows as $row)
    {
        $this->push($row);
    }
}
```
And...
```
$rowsObject = new self($randomRows);
```

Reply:
I like the idea of passing the rows through the constructor as well. 

Would the implementation below be worth considering? It would allow us to avoid creating an argument order dependency. It also provides a bit more flexibility by allowing the rows argument to not be an array which would allow us to shift toward using any object that implements an iterable interface.

```
public function __construct($args = array())
{
     $this->rows = isset($args['rows']) ? $args['rows'] : array();
}
```

Reply:
I don't think I care for `rows` being a key of an array when the constructor only accepts/handles one argument (rows). I *do* agree, however that allowing it to accept any traversable would be a good idea. In that case, it might be something like:
```
public function __construct($rows = array())
{
    // While arrays are traversable with foreach,
    // they will not return true as an instance of Traverable
    if (is_array($rows) || $rows instance of \Traversable)
    {
        foreach ($rows as $row)
        {
            $this->push($row);
        }
    }
}
```
And note that I specifically use `push()` because that method helps ensure the `$row` is a `Relational` model, which other methods of the class rely on. Without that, you could set the internal array to a list of other arrays or `stdClass` objects, etc. and potentially break functionality.